BinaryConverter/src/BinaryConverter.java

```diff
@@ -0,0 +1,366 @@
+					dataType = INT_TYPE;
+				}
+				
+				//now process the numbers....
+				if(dataType == INT_TYPE) {
+					String binInt = "";
+					try {
+						long x = Long.parseLong(nextNumIn);
+						//This is my conversion method...
+						binInt = binConvertInt(x);
+						System.out.println(binInt);
+						
+					} catch (Exception e) {
+						System.out.println(nextNumIn + " -- Number is too large");
+						binInt = "" + nextNumIn + " is too big for Long";
+					}
+					
+				} else if (dataType == FLOAT_TYPE) {
+					//create the 32 bit float method
+				} else {
+					//Strings??
+				}
+			}
+			s. close();
+		}
+	}
+	
+	public static boolean itsNotANumber(String nextNumIn) {
+		boolean itsNotANumber = false;
+		String nonNums = "abcdefghijklmnopqrstuvwxyz!@#$%^&*()_+= {[}]|\\:;\"'<,>?/";
+		
+		for(int l= 0; l < nonNums.length(); l++) {
+			if(nextNumIn.contains("" + nonNums.charAt(l))) {
+				itsNotANumber = true;
+			}
+		}
+		return itsNotANumber;
+	}
+	
+	public static boolean isADecimal(String nextNumIn) {
+		boolean itsADecimal = false;
+		if(nextNumIn.contains(".")) {
+			itsADecimal = true;
+		}
+		return itsADecimal;
+	}
+	
+	
+	
+	public static int findBits(double num) {
+		
+		if(num >= -128 && num <= 127) return 8;
+		if(num >=  -32768 && num <= 32767) return 16;
+		if(num >= -Math.pow(2, 31) && num <= Math.pow(2, 31) - 1) return 32;
+		if(num >= -Math.pow(2, 63) && num <= Math.pow(2, 63) - 1) return 64;
+		
+		return -1;
+	}
+		
+		
+		
+	public static String binConvertDec(double dNum) {
+		
+		String bitString = "";
+		double startNum = dNum;
+		
+		
+		// Now get negative bit
+		String negBit = "";
+		
+		if(dNum >= 0) {
+			negBit = "0";
+		} else {
+			negBit = "1";
+			dNum *= -1;
+		}
+		System.out.println("The negative bit is: " + negBit);
+		
+		
+		//separating int. and dec. part of input number
+		long intNum = (long)dNum;
+		double decNum = dNum - intNum;
+		
+		System.out.println(decNum);
+		
+		
+		//This could be your character array....
+				
+		for(int b = MANTISSA; b >= -MANTISSA; --b)	{	 
+			//System.out.println("b is " + b);
+			//deal with the integer part....
+			if(b >=0) {
+				if( (double)(Math.pow(2.0, (double)b)) <= intNum) {
+					bitString += "1";
+					intNum -= (double)(Math.pow(2.0, (double)b));
+				} else {
+					bitString += "0";
+				}
+				
+			} else {
+				
+				if(b == -1) {
+					bitString += ".";
+				}
+				//decimal part
+				double dnt = 1.0/ (double)(Math.pow(2.0, (double)(b*-1)));
+				if( dnt <= decNum) {
+					bitString += "1";
+					decNum -= dnt;
+				} else {
+					bitString += "0";
+				}
+				
+			}
+
+		}
+		int exponent = 0;
+		String finalScientific = "";
+		String outerTempScientific = "";
+		System.out.println("Bit string is: " + bitString);
+		int firstOne = bitString.indexOf('1');
+		System.out.println("The first 1 is at " + firstOne);
+		int decimalPoint = bitString.indexOf('.');
+		System.out.println("The decimal is at " + decimalPoint);
```

Comment:
Check this part separately. First, check if it is correct mantissa. Then, check if you did the scientific notation correctly.